quantum/process_keycode/process_autocorrect.c

```diff
@@ -25,65 +22,65 @@
  * @return true Continue processing keycodes, and send to host
  * @return false Stop processing keycodes, and don't send to host
  */
-bool process_autocorrection(uint16_t keycode, keyrecord_t* record) {
-    static uint8_t typo_buffer[AUTOCORRECTION_MAX_LENGTH] = {KC_SPC};
-    static uint8_t typo_buffer_size                       = 1;
 
-    if (keycode == AUTO_CTN) {
+bool process_autocorrect(uint16_t keycode, keyrecord_t* record) {
+    static uint8_t typo_buffer[AUTOCORRECT_MAX_LENGTH] = {KC_SPC};
+    static uint8_t typo_buffer_size                    = 1;
+
+    if (keycode == AUTOCRT) {
         if (record->event.pressed) {
             typo_buffer_size = 0;
-            userspace_config.autocorrection ^= 1;
-            eeconfig_update_user(userspace_config.raw);
+            keymap_config.autocorrect_enable ^= 1;
+            eeconfig_update_keymap(keymap_config.raw);
         }
         return false;
     }
 
-    if (!userspace_config.autocorrection) {
+    if (!keymap_config.autocorrect_enable) {
         typo_buffer_size = 0;
         return true;
     }
 
+    if (!record->event.pressed) {
+        return true;
+    }
+
     switch (keycode) {
         case KC_LSFT:
         case KC_RSFT:
             return true;
-#    ifndef NO_ACTION_TAPPING
+#ifndef NO_ACTION_TAPPING
         case QK_MOD_TAP ... QK_MOD_TAP_MAX:
-            if (((keycode >> 8) & 0xF) == MOD_LSFT) {
-                return true;
-            }
-#        ifndef NO_ACTION_LAYER
+#    ifndef NO_ACTION_LAYER
         case QK_LAYER_TAP ... QK_LAYER_TAP_MAX:
-#        endif
-            if (record->event.pressed || !record->tap.count) {
-                return true;
+#    endif
+            if (!record->tap.count) {
+                return true;  // Early return if tap-hold key is not being tapped.
             }
             keycode &= 0xFF;
             break;
-#    endif
-#    ifdef SWAP_HANDS_ENABLE
+#endif
+#ifdef SWAP_HANDS_ENABLE
         case QK_SWAP_HANDS ... QK_SWAP_HANDS_MAX:
-            if (keycode >= 0x56F0 || record->event.pressed || !record->tap.count) {
+            if (keycode >= 0x56F0 || !record->tap.count) {
                 return true;
             }
             keycode &= 0xFF;
             break;
-#    endif
-#    ifndef NO_ACTION_ONESHOT
+#endif
+#ifndef NO_ACTION_ONESHOT
         case QK_ONE_SHOT_MOD ... QK_ONE_SHOT_MOD_MAX:
             if ((keycode & 0xF) == MOD_LSFT) {
                 return true;
             }
-#    endif
+            // fall through intentionally
+#endif
         default:
```

Comment:
Another case worth adding is for shifted keycodes `S(kc)`:

```c
case QK_LSFT ... QK_LSFT + 255:
case QK_RSFT ... QK_RSFT + 255:
  keycode &= 0xFF;  // Get the basic keycode.
  break;
```

For instance, this covers the shifted symbol KC_EXLM = S(KC_1). Without this case, the logic below won't recognize KC_EXLM and clear the autocorrect state. But with this case, keycode is changed to KC_1, and the logic below will treat it as a word breaking char as it should be.